VMS.Application/Services/ActivityService.cs

```diff
@@ -31,25 +31,76 @@ public class ActivityService : BaseService, IActivityService
             _addressLocationService = addressLocationService;
         }
 
-
-        public async Task<List<ActivityViewModel>> GetAllActivitiesAsync()
+        public async Task<List<ActivityViewModel>> GetAllActivitiesAsync(bool isSearch, string searchValue, FilterActivityViewModel filter)
         {
             DbContext dbContext = _dbContextFactory.CreateDbContext();
+            if (isSearch)
+            {
+                return await GetAllActivitiesWithSearchValueAsync(searchValue, dbContext);
+            }
+            else
+            {
+                return await GetAllActivitiesWithFilterAsync(filter, dbContext);
+            }
+        }
+
+        private async Task<List<ActivityViewModel>> GetAllActivitiesWithSearchValueAsync(string searchValue, DbContext dbContext)
+        {
+            Specification<Activity> specification = new()
+            {
+                Conditions = new List<Expression<Func<Activity, bool>>>()
+                {
+                    a => a.Name.ToUpper().Trim().Contains(searchValue.ToUpper().Trim())
+                }
+            };
```

Comment:
Another possible condition is that an activity has not ended yet.
Please add this condition to other appropriate places.

Reply:
oh, i miss that, my bad